sdk-java/src/main/java/io/littlehorse/sdk/wfsdk/internal/WorkflowThreadImpl.java

```diff
@@ -519,6 +519,21 @@ public void addTimeoutToExtEvt(NodeOutputImpl node, int timeoutSeconds) {
         spec.putNodes(node.nodeName, n.build());
     }
 
+    public void addFailureHandlerOnWaitForThreadsNode(WaitForThreadsNodeOutputImpl node, FailureHandlerDef handler) {
+        checkIfIsActive();
+        Node.Builder n = spec.getNodesOrThrow(node.nodeName).toBuilder();
+
+        if (n.getNodeCase() != NodeCase.WAIT_FOR_THREADS) {
+            throw new IllegalStateException("orzdash this should only be a WAIT_FOR_THREADS node");
```

Comment:
Suggestion:
`throw new IllegalStateException("Only WAIT_FOR_THREADS nodes can have child exception handlers.");`

Reply:
It should be impossible for a user to make this happen; otherwise, it means that the code I wrote doesn't behave how I thought it does. I will make a note of that.